### PR TITLE
Tickets/dm 29841

### PR DESCRIPTION
**Pull request by jeffcarlin**
This branch came about because of changes in how external calibrations (e.g., jointcal WCS and FGCM photoCalibs) are applied in the stack. Major changes:

- Add the (keyword-based) switching in `MatchedCatalogBase.py` to select between connections depending on whether external calibrations or "defaults" are requested.
- Select appropriate `photoCalib` and `astromCalib` for each dataId in the `run` method of `MatchedCatalogBase`, so that each dataId has its associated calibs passed along with it to the matcher. This avoids looking up calibs (or implementing logic to think about how to use/find the correct calibs) within the matching routine, and ensures that the matcher is always being passed one calib per dataId.
- Convert (some of the) variables within `MatchedCatalogBase` and `matcher.py` to camelCase rather than snake_case.
- Use `dataId` instead of `vId` for clarity.

NOTE: these changes are only for the "matched catalog" metric calculations. Other types of metrics will be updated to use the new calibration format on separate tickets.

### CODE REVIEW

File: pipelines/preparation/preparation_matched_no_extcalibs.yaml
Comment (bechtol) [inline pipelines/preparation/preparation_matched_no_extcalibs.yaml:7]: Previously, we have used a convention to not set the configuration in the pipeline file when the default values are desired.

File: pipelines/preparation/preparation_matched_jointcal_fgcm_tractonly.yaml
Comment (bechtol) [inline pipelines/preparation/preparation_matched_jointcal_fgcm_tractonly.yaml:7]: I expected the following here in order to apply the external calibrations:
```
        config.doApplyExternalPhotoCalib = True
        config.useGlobalExternalPhotoCalib = True
        config.doApplyExternalSkyWcs = True
        config.useGlobalExternalSkyWcs = False
```

File: python/lsst/faro/base/MatchedCatalogBase.py
Comment (bechtol) [inline python/lsst/faro/base/MatchedCatalogBase.py:124]: As long as we are making change to camelCase for consistency, it seems that this function should be matchCatalogs

File: python/lsst/faro/utils/matcher.py
Comment (bechtol) [inline python/lsst/faro/utils/matcher.py:12]: camelCase for matchCatalogs for consistency

File: python/lsst/faro/utils/matcher.py
Comment (bechtol) [inline python/lsst/faro/utils/matcher.py:153]: camelCase for consistency

**Review by bechtol — APPROVED** [review]
Thanks @jeffcarlin for making these updates. Looks good.

See the few minor name convention comments. I expected different configuration choices in `preparation_matched_jointcal_fgcm_tractonly.yaml` to apply the external calibrations.

File: pipelines/preparation/preparation_matched_no_extcalibs.yaml
Comment (jeffcarlin) [reply pipelines/preparation/preparation_matched_no_extcalibs.yaml:7]: Good point. Removing those configs would make this file exactly the same as `preparation_matched.yaml`, so I just removed it.

File: pipelines/preparation/preparation_matched_jointcal_fgcm_tractonly.yaml
Comment (jeffcarlin) [reply pipelines/preparation/preparation_matched_jointcal_fgcm_tractonly.yaml:7]: Yes, should have been (I had changed it for testing, and forgot to revert).

**Comment by timj** [comment]
@jeffcarlin  this PR also seems to have a bad merge in it:

> @jeffcarlin Merge branch 'tickets/DM-29841' of https://github.com/lsst/faro into tickets/DM-29841
 
Please rebase before merging. Do you want me to give you a training session?

cc/ @ktlim

**Comment by jeffcarlin** [comment]
Yes, it has some ugly history _because_ of the rebase that I did. I am a git novice and did a poor job of rebasing.

**Comment by timj** [comment]
Okay. Next time if you know you have a bad commit in the history but don't know how to fix it, please ask for help. Do not merge it anyhow.

**Comment by jeffcarlin** [comment]
I think I don't know what a "bad commit" even means. This one got messy because there were lots of merge conflicts when rebasing.

**Comment by timj** [comment]
If you see "merge branch x of http x" that is a red flag. Any merge commits on your branch is a red flag unless you know you are explicitly using an integration branch with many different tickets going on to it before master.

**Pull request merged**
